tasks/main.yml

```diff
@@ -0,0 +1,70 @@
+---
+# Override some variables
+- set_fact:
+    kafka_stunnel_backend_hosts: "{{kafka_hosts|regex_replace(':.*',':{{kafka_stunnel_server_port}}')}}"
+    kafka_hosts: 127.0.0.1:{{kafka_stunnel_client_port}}
+    is_stunnel_enabled: true
```

Comment:
The rewriting of a variable is a bit of a surprising behavior. It is elegant because kafka_hosts can be set for the entire cluster and then each box ends up talking specifically to its proper tunnel. However the behavior is surprising and can be easy to undermine. If any of the roles using this run without this playbook run first they will reconfigure without stunnel. That means you can't call out individual roles by tag, something I commonly do. Additionally I believe (but am not sure) that variables reset between plays of a single playbook. That means you need this role run for each play. It is common to separate plays for example in the test clusters zookeeper runs in an earlier play so kafka and kafka topics can be built before starting up monasca components. Even amoung monasca components thresh runs in its own play as it is targeted at only the master node. You should verify if variables reset across plays in the same run.

Perhaps a less surprising way for this work would be to utilize [fact caching](http://docs.ansible.com/playbooks_variables.html#fact-caching). This role could then register stunnel_kafka_hosts and cache that fact and the roles needing kafka_hosts pull it from the fact cache. This would allow the relationship to be clear to anyone looking at variables as well as allow using tags, and separate plays.

Either way the readme could use a more complete explanation of what is happening.